modules/oe_theme_content_event/src/Plugin/ExtraField/Display/LivestreamExtraField.php

```diff
@@ -0,0 +1,114 @@
+
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Core\Datetime\DateFormatterInterface;
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\oe_content_event\EventNodeWrapper;
+use Drupal\oe_time_caching\Cache\TimeBasedCacheTagGeneratorInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Extra field displaying livestream information on events.
+ *
+ * @ExtraFieldDisplay(
+ *   id = "oe_theme_content_event_livestream",
+ *   label = @Translation("Livestream"),
+ *   bundles = {
+ *     "node.oe_event",
+ *   },
+ *   visible = true
+ * )
+ */
+class LivestreamExtraField extends DateAwareExtraFieldBase {
+
+  /**
+   * Date formatter service.
+   *
+   * @var \Drupal\Core\Datetime\DateFormatterInterface
+   */
+  protected $dateFormatter;
+
+  /**
+   * LivestreamExtraField constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Component\Datetime\TimeInterface $time
+   *   The time service.
+   * @param \Drupal\oe_time_caching\Cache\TimeBasedCacheTagGeneratorInterface $cache_tag_generator
+   *   Time based cache tag generator service.
+   * @param \Drupal\Core\Datetime\DateFormatterInterface $date_formatter
+   *   The date formatter.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager, TimeInterface $time, TimeBasedCacheTagGeneratorInterface $cache_tag_generator, DateFormatterInterface $date_formatter) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $entity_type_manager, $time, $cache_tag_generator);
+    $this->dateFormatter = $date_formatter;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('datetime.time'),
+      $container->get('oe_time_caching.time_based_cache_tag_generator'),
+      $container->get('date.formatter')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getLabel() {
+    return $this->t('Livestream');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(ContentEntityInterface $entity) {
+    $event = EventNodeWrapper::getInstance($entity);
+
+    if (!$event->hasOnlineType()
+      || !$event->hasOnlineLink()
+      || !$event->hasOnlineDates()
+      || !$event->isOnlinePeriodYetToCome($this->requestDateTime)) {
```

Comment:
It's ok to return empty if type, link and dates are empty (as, due to custom validation logic, filling one requires the other two to be filled as well), but we should isolate the isOnlinePeriodYetToCome check, so that we apply cache tags when type, link and dates are filled, but it's still not time to show the field.